### PR TITLE
[SPARK-31753][SQL][DOCS][FOLLOW-UP] Add missing keywords in the SQL docs

### PR DESCRIPTION
**Pull request by GuoPhilipse**
### What changes were proposed in this pull request?
update sql-ref docs, the following key words will be added in this PR.

CLUSTERED BY 
SORTED BY
INTO num_buckets BUCKETS

### Why are the changes needed?
let more users know the sql key words usage

### Does this PR introduce _any_ user-facing change?
No
image
image
image

### How was this patch tested?
generate html test

### CODE REVIEW

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md [hunk 1]
```diff
@@ -65,6 +68,18 @@ as any order. For example, you can write COMMENT table_comment after TBLPROPERTI
 
     Partitions are created on the table, based on the columns specified.
     
+* **CLUSTERED BY**
+
+    Specifies bucket columns for bucketing table.
+    
+* **SORTED BY**
+
+    Used to sort bucket column, we can combine with `ASC` for ascending order, with `DESC` for descending order.
```
Comment (maropu) [inline docs/sql-ref-syntax-ddl-create-table-hiveformat.md:77]: How about rephrasing it like this? `Specifies an ordering of bucket columns. Optionally, one can use ASC for an ascending order or DESC  for a descending order after any column names in the SORTED BY clause. If not specified, ASC is assumed by default.`

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md [hunk 2]
```diff
@@ -65,6 +68,18 @@ as any order. For example, you can write COMMENT table_comment after TBLPROPERTI
...
+    
+* **INTO num_buckets BUCKETS**
+
+    Specifies buckets numbers, which is used in  `CLUSTERED BY` clause.
```
Comment (maropu) [inline docs/sql-ref-syntax-ddl-create-table-hiveformat.md:81]: nit: redundant spaces found between `in`/`CLUSTER BY`. And, `in the CLUSTERED BY clause`?

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md [hunk 3]
```diff
@@ -203,6 +218,17 @@ CREATE EXTERNAL TABLE family (id INT, name STRING)
     STORED AS INPUTFORMAT 'com.ly.spark.example.serde.io.SerDeExampleInputFormat'
         OUTPUTFORMAT 'com.ly.spark.example.serde.io.SerDeExampleOutputFormat'
     LOCATION '/tmp/family/';
+
+--Use `CLUSTERED BY` clause to create bucket table without `SORTED BY`
+CREATE TABLE TEST1(ID INT, AGE STRING)
```
Comment (maropu) [inline docs/sql-ref-syntax-ddl-create-table-hiveformat.md:223]: nit: To follow the format of the other examples, `CREATE TABLE clustered_by_test1 (ID ...`?

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md [hunk 4]
```diff
@@ -203,6 +218,17 @@ CREATE EXTERNAL TABLE family (id INT, name STRING)
...
+    CLUSTERED BY (ID)
+    INTO 4 BUCKETS
+
+--Use `CLUSTERED BY` clause to create bucket table with `SORTED BY`
+CREATE TABLE TEST2(ID INT, NAME STRING)
```
Comment (maropu) [inline docs/sql-ref-syntax-ddl-create-table-hiveformat.md:228]: ditto

**Comment by maropu** [comment]
ok to test

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md [hunk 5]
```diff
@@ -65,6 +68,18 @@ as any order. For example, you can write COMMENT table_comment after TBLPROPERTI
 
     Partitions are created on the table, based on the columns specified.
     
+* **CLUSTERED BY**
+
+    Specifies bucket columns for bucketing table.
```
[the remark this answers is not in the document]
Comment (huaxingao) [reply docs/sql-ref-syntax-ddl-create-table-hiveformat.md:73]: Should we use the same description as the one in ```sql-ref-syntax-ddl-create-table-datasource```?

[the remark this answers is not in the document]
Comment (GuoPhilipse) [reply docs/sql-ref-syntax-ddl-create-table-hiveformat.md:73]: emm, sorry,  i miss that,  let me keep pace with the one in `sql-ref-syntax-ddl-create-table-datasource`

**Comment by maropu** [comment]
Could you add the screenshots of the updated docs in the PR description? LGTM otherwise.

**Comment by GuoPhilipse** [comment]
> Could you add the screenshots of the updated docs in the PR description? LGTM otherwise.
@maropu Have updated the html screenshots :)

**Review by maropu — APPROVED** [review]
cc: @huaxingao @gatorsmile

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md [hunk 6]
```diff
@@ -31,6 +31,9 @@ CREATE [ EXTERNAL ] TABLE [ IF NOT EXISTS ] table_identifier
     [ COMMENT table_comment ]
     [ PARTITIONED BY ( col_name2[:] col_type2 [ COMMENT col_comment2 ], ... ) 
         | ( col_name1, col_name2, ... ) ]
+    [ CLUSTERED BY ( col_name1, col_name2, ...)
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-create-table-hiveformat.md:34]: super nit: add a space between `...` and `)`

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md [hunk 7]
```diff
@@ -203,6 +221,20 @@ CREATE EXTERNAL TABLE family (id INT, name STRING)
...
+--Use `CLUSTERED BY` clause to create bucket table without `SORTED BY`
+CREATE TABLE clustered_by_test1 (ID INT, AGE STRING)
+    CLUSTERED BY (ID)
+    INTO 4 BUCKETS
+    STORED AS ORC
+
+--Use `CLUSTERED BY` clause to create bucket table with `SORTED BY`
+CREATE TABLE clustered_by_test2 (ID INT, NAME STRING)
+    PARTITIONED BY (YEAR STRING)
+    CLUSTERED BY (ID,NAME)
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-create-table-hiveformat.md:234]: super nit: add a space after `,`

**Review by huaxingao — APPROVED** [review]

**Comment by GuoPhilipse** [comment]
Thanks@maropu @huaxingao ,have updated :)

**Comment by maropu** [comment]
Thanks! Merged to master/3.0.

**Pull request closed without merging**